scripts/install_go.sh

```diff
@@ -0,0 +1,146 @@
+
+OS="$(uname -s)"
+ARCH="$(uname -m)"
+
+case $OS in
+    "Linux")
+        case $ARCH in
+        "x86_64")
+            ARCH=amd64
+            ;;
+        "aarch64")
+            ARCH=arm64
+            ;;
+        "armv6")
+            ARCH=armv6l
+            ;;
+        "armv8")
+            ARCH=arm64
+            ;;
+        .*386.*)
+            ARCH=386
+            ;;
+        esac
+        PLATFORM="linux-$ARCH"
+    ;;
+    "Darwin")
+        PLATFORM="darwin-amd64"
+    ;;
+esac
+
+print_help() {
+    echo "Usage: bash goinstall.sh [OPTIONS]"
+    echo -e "\nOPTIONS:"
+    echo -e "  --remove\tRemove currently installed version"
+    echo -e "  --version\tSpecify a version number to install"
+}
+
+if [ -n "`$SHELL -c 'echo $ZSH_VERSION'`" ]; then
+    shell_profile="zshrc"
+elif [ -n "`$SHELL -c 'echo $BASH_VERSION'`" ]; then
+    shell_profile="bashrc"
+fi
+
+if [ "$1" == "--remove" ]; then
+    sudo rm -rf "$GOROOT"
+    if [ "$OS" == "Darwin" ]; then
+        sed -i "" '/# GoLang/d' "$HOME/.${shell_profile}"
+        sed -i "" '/export GOROOT/d' "$HOME/.${shell_profile}"
+        sed -i "" '/$GOROOT\/bin/d' "$HOME/.${shell_profile}"
+        sed -i "" '/export GOPATH/d' "$HOME/.${shell_profile}"
+        sed -i "" '/$GOPATH\/bin/d' "$HOME/.${shell_profile}"
+    else
+        sed -i '/# GoLang/d' "$HOME/.${shell_profile}"
+        sed -i '/export GOROOT/d' "$HOME/.${shell_profile}"
+        sed -i '/$GOROOT\/bin/d' "$HOME/.${shell_profile}"
+        sed -i '/export GOPATH/d' "$HOME/.${shell_profile}"
+        sed -i '/$GOPATH\/bin/d' "$HOME/.${shell_profile}"
+    fi
+    echo "Go removed."
+    exit 0
+elif [ "$1" == "--help" ]; then
+    print_help
+    exit 0
+elif [ "$1" == "--version" ]; then
+    if [ -z "$2" ]; then # Check if --version has a second positional parameter
+        echo "Please provide a version number for: $1"
+    else
+        VERSION=$2
+    fi
+elif [ ! -z "$1" ]; then
+    echo "Unrecognized option: $1"
+    exit 1
+fi
+
+if [ -d "$GOROOT" ]; then
+    echo "The Go install directory ($GOROOT) already exists. Exiting."
+    exit 1
+fi
+
+PACKAGE_NAME="go$VERSION.$PLATFORM.tar.gz"
+TEMP_DIRECTORY=$(mktemp -d)
+
+echo "Downloading $PACKAGE_NAME ..."
+if hash wget 2>/dev/null; then
+    wget https://golang.org/dl/$PACKAGE_NAME -O "$TEMP_DIRECTORY/go.tar.gz"
+else
+    curl -o "$TEMP_DIRECTORY/go.tar.gz" https://golang.org/dl/$PACKAGE_NAME
+fi
+
+if [ $? -ne 0 ]; then
+    echo "Download failed! Exiting."
+    exit 1
+fi
+
+echo "Extracting File..."
+sudo mkdir -p "$GOROOT"
+sudo tar -C "$GOROOT" --strip-components=1 -xzf "$TEMP_DIRECTORY/go.tar.gz"
+touch "$HOME/.${shell_profile}"
+{
+    echo '# GoLang'
+    echo "export GOROOT=${GOROOT}"
+    echo 'export PATH=$GOROOT/bin:$PATH'
+    echo "export GOPATH=$GOPATH"
+    echo 'export PATH=$GOPATH/bin:$PATH'
+} >> "$HOME/.${shell_profile}"
+
+mkdir -p $GOPATH/{src,pkg,bin}
+source $HOME/.${shell_profile}
+rm -f "$TEMP_DIRECTORY/go.tar.gz"
```

Comment:
I'm wondering whether these this script should be placed within a `Dockerfile` and not provided as an individual installation script.  The reason for this is that we also don't have scripts to build/install other dependencies, such as python.

Reply:
Compared to python, go is not very widespread and installation can be difficult for new users... The idea is to automatically setup the new environment without any user interaction.

Reply:
Compared to building unikernels, go is relatively simply to install.  Besides, this installation and subsequent compilation of programs can still be provided in the `Dockerfile` for kraft such that the executables are distributed only and not the full go installation.